src/zope/globalrequest/tests.py

```diff
@@ -1,37 +1,93 @@
-from unittest import TestSuite
-from zope.configuration import config
-from zope.configuration import xmlconfig
-from zope.testing.cleanup import cleanUp
-import doctest
-import zope.app.wsgi.testlayer
-import zope.globalrequest
-import zope.testbrowser.wsgi
+# -*- coding: utf-8 -*-
+import unittest
 
 
-def zcml(source):
-    context = config.ConfigurationMachine()
-    xmlconfig.registerCommonDirectives(context)
-    xmlconfig.string(source, context)
+class TestGlobalrequest(unittest.TestCase):
 
+    def tearDown(self):
+        # reset the threading local context
+        from threading import local
+        import zope.globalrequest.local
```

Comment:
Well, its been a while I had those and its probably not that interesting. Its primary circular imports or importing code early so possible errors are popping up on load time of the test and not on execution time, when you would like to expect them.